.github/workflows/docs.yaml

```diff
@@ -1,8 +1,25 @@
 name: Docs
 
 on:
+  pull_request:
+    paths:
+      - "docs/**"
+      - "**.md"
+      - "**.rst"
+      - ".github/workflows/*"
+      - "!.github/workflows/docs.yaml"
   push:
-    branches: ["main"]
+    paths:
+      - "docs/**"
+      - "**.md"
+      - "**.rst"
+      - ".github/workflows/*"
+      - "!.github/workflows/docs.yaml"
+    branches-ignore:
+      - "dependabot/**"
+      - "pre-commit-ci-update-config"
```

Comment:
pre-commit.ci / dependabot pushes to these branches and opens a PR. So like this one would avoid triggering two workflow executions - one for the pushed branch, one for the PR.